src/components/datepicker/js/datepickerDirective.js

```diff
@@ -338,6 +330,25 @@
     /** @type {Number} Extra margin for the top of the floating calendar. Gets determined on the first open. */
     this.topMargin = null;
 
+    /** @type {!MdPanelPosition} Position object for the mdPanel instance. */
+    this.panelPosition = $mdPanel.newPanelPosition()
+      .relativeTo(this.inputContainer)
+      .addPanelPosition($mdPanel.xPosition.ALIGN_START, $mdPanel.yPosition.ALIGN_TOPS)
+      .withOffsetX(-this.leftMargin + 'px')
+      .withOffsetY(angular.bind(this, this.getPanelYOffset));
+
+    /** @type {!MdPanelRef} Reference to the mdPanel instance of the calendar. */
+    this.panelRef = $mdPanel.create({
+      attachTo: document.body,
+      template: CALENDAR,
+      clickOutsideToClose: true,
+      escapeToClose: true,
+      focusOnOpen: false,
+      scope: this.$scope.$new(),
```

Comment:
The default creates a scope that's a child of `$rootScope`, however we need it to be a child of the datepicker's scope.